File: core/src/main/java/com/picker/imagepickerwithcrop/ImagePickerManager.java
package com.picker.imagepickerwithcrop;

import android.app.Activity;
import android.content.Intent;
import android.net.Uri;
import android.provider.MediaStore;

/**
 * Created by Mickael on 10/10/2016.
 */

//TODO convert kotlin

public class ImagePickerManager extends PickerManager {

    public ImagePickerManager(Activity activity) {
        super(activity);
    }

    protected void sendToExternalApp() {
        final Intent intent = new Intent(Intent.ACTION_PICK, MediaStore.Images.Media.EXTERNAL_CONTENT_URI);
        intent.setType("image/*");

        activity.startActivityForResult(Intent.createChooser(intent, "Select avatar..."),
                REQUEST_CODE_SELECT_IMAGE);
    }

    @Override
    public void setUri(Uri uri) {
        mProcessingPhotoUri = uri;
    }

}

File: core/src/main/java/com/picker/imagepickerwithcrop/ImagePickerManager.kt
package com.picker.imagepickerwithcrop

import android.app.Activity
import android.content.Intent
import android.net.Uri
import android.provider.MediaStore

class ImagePickerManager(
        activity: Activity
) : PickerManager(activity) {

    override fun sendToExternalApp() {
        val intent = Intent(Intent.ACTION_PICK, MediaStore.Images.Media.EXTERNAL_CONTENT_URI)
        intent.type = "image/*"
        activity.startActivityForResult(Intent.createChooser(intent, "Select avatar..."), REQUEST_CODE_SELECT_IMAGE)
    }

    override fun setUri(uri: Uri?) {
        super.setUri(uri)

        mProcessingPhotoUri = uri
    }
}
